code/modules/power/supermatter/supermatter.dm

```diff
@@ -537,7 +537,7 @@
 
 	//Makes em go mad and accumulate rads.
 	for(var/mob/living/carbon/human/l in view(src, HALLUCINATION_RANGE(power))) // If they can see it without mesons on.  Bad on them.
-		if(!istype(l.glasses, /obj/item/clothing/glasses/meson) && !HAS_TRAIT(l, TRAIT_MESON_VISION))
+		if(!istype(l.glasses, /obj/item/clothing/glasses/meson) && !HAS_TRAIT(l, TRAIT_MESON_VISION) && !istype(l.get_int_organ(/obj/item/organ/internal/eyes), /obj/item/organ/internal/eyes/cybernetic/meson))
```

Comment:
```suggestion
		if(!istype(l.glasses, /obj/item/clothing/glasses/meson) && !HAS_TRAIT(l, TRAIT_MESON_VISION) && !l.get_int_organ(/obj/item/organ/internal/eyes/cybernetic/meson)
```
because `get_int_organ` returns the organ that you are checking for, there is no need to do an `istype` check,  you can just check directly for the organ type you want.